sdk/src/account_meta.rs

```diff
@@ -0,0 +1,117 @@
+//! Types used
+
+use anchor_lang::{AnchorDeserialize, AnchorSerialize};
+use solana_program::pubkey::Pubkey;
+
+use crate::{
+    compressed_account::CompressedAccountWithMerkleContext,
+    merkle_context::{
+        pack_address_merkle_context, pack_merkle_context, AddressMerkleContext,
+        PackedAddressMerkleContext, PackedMerkleContext, RemainingAccounts,
+    },
+};
+
+#[derive(Debug, Clone, AnchorSerialize, AnchorDeserialize, PartialEq, Default)]
+pub struct LightAccountMeta {
+    /// Lamports.
+    pub lamports: Option<u64>,
+    /// Address of the account (the address can change).
+    pub address: Option<[u8; 32]>,
+    /// Data of the account.
+    pub data: Option<Vec<u8>>,
+    /// Merkle tree.
+    pub merkle_context: Option<PackedMerkleContext>,
+    /// Merkle tree root index.
+    pub merkle_tree_root_index: Option<u16>,
+    /// Output Merkle tree.
+    pub output_merkle_tree_index: Option<u8>,
+    /// Address Merkle tree. Set only when adding or updating the address.
+    pub address_merkle_context: Option<PackedAddressMerkleContext>,
+    /// Address Merkle tree root index. Set only when adding or updating the
+    /// address.
+    pub address_merkle_tree_root_index: Option<u16>,
+    /// Account is read only.
+    /// (not used for now, just a placeholder)
+    pub read_only: bool,
+}
+
+impl LightAccountMeta {
+    #[allow(clippy::too_many_arguments)]
+    pub fn new_init(
+        data: Option<Vec<u8>>,
```

Comment:
I think we should remove this, since a new account doesn't have any existing data and new data should be set onchain so that it can be trusted.

Reply:
Done